grading-container/script/grading_job/grading_job.py

```diff
@@ -21,20 +28,75 @@ class GradingJob:
+    """
+    Executes a grading job in full with the following workflow:
+      1.) Download student, as well as Possibly starter and professor code.
+      2.) Execute this object's GradingScript
+      3.) Return the results in a GradingJobOutput object.
+    """
+    audit.log_details(f"Executing grading job with grade_id {self.__grade_id} "\
+      "and submission_id {self.__submission_id}...")
+    build_dir = f"{secret}_build"
+    self.__interpolated_dirs["$BUILD"] = build_dir
+    self.retrieve_job_files()
+    # TODO: Replace with actual object.
+    output: GradingJobOutput = None
+    if output.has_tap_output():
+      audit.log_details("Successfully completed running the grading script.")
+    else: 
+      audit.log_details("Script exeuction was aborted due to a number of errors.")
+    audit.log_details("Grading job complete.")
+    output.set_audit_log(audit.get_complete_log())
+    return output
+
+  def retrieve_job_files(self, retrieval_cmd_factory: FileRetrievalCommandFactory, 
+    audit: Audit, secret: str) -> None:
+    """
+    Retrieves the code files necessary to run this GradingJob. This method has many side effects, 
+    including the following:
+      - Downloading either individual code files (e.g., code.java) or .zip files.
+      - Placing files (i.e., extraction in case of .zip) into specific directories.
+      - Updating the interpolated directories dictionary.
+    """
+    audit.log_details(f"Retrieving student files from {self.__student_code}...")
+    student_rfc = retrieval_cmd_factory.get_file_retrieval_command(self.__student_code)
+    student_rfc.execute()
+    self.__interpolated_dirs["$STUDENT"] = student_rfc.get_save_dir()
+    audit.log_details("Successfully retrieved student file(s).")
+    # Both of these must exist.
+    if self.__starter_code and self.__professor_code:
+      # Retrieve Starter Code
+      audit.log_details(f"Retrieving starter code from {self.__starter_code}")
+      starter_rfc = retrieval_cmd_factory.get_file_retrieval_command(self.__starter_code)
+      starter_rfc.execute()
+      self.__interpolated_dirs["$STARTER"] = starter_rfc.get_save_dir()
+      audit.log_details("Successfully retrieved starter code.")
+      # Retrieve Professor Code/Tests
+      audit.log_details(f"Retrieving professor tests/code from {self.__professor_code}")
+      professor_rfc = retrieval_cmd_factory.get_file_retrieval_command(self.__professor_code)
+      professor_rfc.execute()
+      self.__interpolated_dirs["$PROFESSOR"] = professor_rfc.get_save_dir()
+      audit.log_details("Successfully retrieved professor tests/code.")
+    
   @staticmethod
   def generate_grading_script(commands: List[GradingScriptCommandJSON], max_retries: int = DEFAULT_NUM_RETRIES):
     """
     Given a list of grading script commands, generate a grading script
     that can take these commands and execute them.
     """
     def cmd_json_to_class(cmd: GradingScriptCommandJSON) -> GradingScriptCommand:
```

Comment:
This seems...pretty incomplete.  We had talked about using the on-complete and on-fail fields to build up a DAG of commands that you'd work through, but it doesn't seem like that's been built yet, correct?